enhancements/api-review/add-new-CRD-ImageContentPolicy(ICP)-to-config.openshift.io.md

```diff
@@ -0,0 +1,260 @@
+
+## Proposal
+
+[openshift/api#874:Create ImageContentPolicy CRD to config/v1 and allowMirrorByTags support](https://github.com/openshift/api/pull/874) PR was merged. New CRD ImageContentPolicy was added to config.openshift.io/v1 PR.
+
+The schema of ImageContentPolicy(ICP) = fields of ImageContentSourcePolicy(ICSP) + allowMirrorByTags
+
+The `allowMirrorByTags` (default false) property will allow for mirroring by-tag images through configured mirrors.
+
+```go
+// ImageContentPolicySpec is the specification of the ImageContentPolicy CRD.
+type ImageContentPolicySpec struct {
+	// repositoryDigestMirrors allows images referenced by image digests in pods to be
+	// pulled from alternative mirrored repository locations. The image pull specification
+	// provided to the pod will be compared to the source locations described in RepositoryDigestMirrors
+	// and the image may be pulled down from any of the mirrors in the list instead of the
+	// specified repository allowing administrators to choose a potentially faster mirror.
+	// To pull image from mirrors by tags, should set the "allowMirrorByTags".
+	//
+	// If the image pull specification matches the reposirtory of "source" in multiple policies,
+	// only the policies have the most specific namespace matching will be used.
+	// For example, if there are policies using quay.io/libpod and quay.io/libpod/busybox as
+	// the "source", only the policies using quay.io/libpod/busybox are going to apply
+	// for pull specification quay.io/libpod/busybox.
+	// Each “source” repository is treated independently; configurations for different “source”
+	// repositories don’t interact.
+	//
+	// If the "mirrors" is not specified, the image will continue to be pulled from the specified
+	// repository in the pull spec.
+	//
+	// When multiple policies are defined for the same “source” repository, the sets of defined
+	// mirrors will be merged together, preserving the relative order of the mirrors, if possible.
+	// For example, if policy A has mirrors `a, b, c` and policy B has mirrors `c, d, e`, the
+	// mirrors will be used in the order `a, b, c, d, e`.  If the orders of mirror entries conflict
+	// (e.g. `a, b` vs. `b, a`) the configuration is not rejected but the resulting order is unspecified.
+	// +optional
+	// +listType=atomic
+	RepositoryDigestMirrors []RepositoryDigestMirrors `json:"repositoryDigestMirrors"`
+}
+
+// RepositoryDigestMirrors holds cluster-wide information about how to handle mirrors in the registries config.
+type RepositoryDigestMirrors struct {
+	// source is the repository that users refer to, e.g. in image pull specifications.
+	// "source" ueses one of the following formats:
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// host[:port]/namespace[/namespace…]/repo(:_tag|@digest)
```

Comment:
Unlike containers-registries.conf(5), https://github.com/openshift/runtime-utils/blob/8b8348d80d1d1e7b6cf06fb009d5965e0b55baa2/pkg/registries/registries.go#L19 doesn’t currently support the `:tag|@digest` part; it can be a `host[:port][[/namespace…]/repo]` only.

(I don’t think single-image mirrors are worth worrying about… unless we needed to allow them now because we won’t be allowed to change the validation rules again?)